llvm/lib/Target/PowerPC/PPCFrameLowering.cpp

```diff
@@ -376,6 +376,9 @@ bool PPCFrameLowering::needsFP(const MachineFunction &MF) const {
   if (MF.getFunction().hasFnAttribute(Attribute::Naked))
     return false;
 
+  if (Subtarget.isAIXABI() && Subtarget.getRegisterInfo()->hasBasePointer(MF))
+    return true;
```

Comment:
hmm, for this case, what we need is the backup/restore instructions for r31 because r30 is used, not the real frame pointer register, i.e. if we change the API here, we have redundant `mr 31, 1`. Can we just add the backup/restore instructions instead? For example, just change `emitPrologue()` and `emitEpilogue` for the store/load? (May also need to change the place where the stack object is allocated for the frame pointer register.)